### PR TITLE
Fix video4linux headers

### PR DESCRIPTION
**Pull request by mcb30**
The `linux/videodev.h` header was removed in kernel 2.6.38.  A drop-in replacement header is available from libv4l as `libv4l1-videodev.h`: use that if available.

Unlike the original `linux/videodev.h` header, the replacement `libv4l1-videodev.h` header does not automatically include `linux/videodev2.h`: include `linux/videodev2.h` explicitly as needed.

### CODE REVIEW

File: src/v4l.c [hunk 1]
```diff
@@ -41,11 +41,18 @@ so, delete this exception statement from your version.
...
 #define CONFIG_VIDEO_V4L1_COMPAT
+#if HAVE_LIBV4L1_VIDEODEV_H
+#include <libv4l1-videodev.h>
+#if HAVE_LINUX_VIDEODEV2_H
+#include <linux/videodev2.h>
+#endif
+#else
 #include <linux/videodev.h>
+#endif
 #ifdef __LINUX_VIDEODEV2_H
```
Comment (bk138) [inline src/v4l.c:56]: Is this one related to the check in line 50?

File: src/v4l.c [hunk 2]
```diff
@@ -41,11 +41,18 @@ so, delete this exception statement from your version.
 #include "keyboard.h"
 #include "allowed_input_t.h"
 
-#if HAVE_LINUX_VIDEODEV_H
+#if HAVE_LIBV4L1_VIDEODEV_H || HAVE_LINUX_VIDEODEV_H
```
Comment (bk138) [inline src/v4l.c:44]: TBH, those checks look quite convoluted now - you're free to throw the current status quo away and do a new implementation that checks for v4l1 and v4l2, respectively.

Comment (mcb30) [reply src/v4l.c:44]: Thanks - that would definitely make it possible to simplify the code.  Since v4l1 is no longer supported by kernel 2.6.38 and later, can we ditch it completely and support only v4l2?

Comment (bk138) [reply src/v4l.c:44]: As the project is more or less in maintenance mode, please try to make a change that as noninvasive as possible. I guess cleaning up the header check would suffice.

Comment (mcb30) [reply src/v4l.c:44]: Thanks.  I'll leave the PR as-is in that case.  It would take a moderately invasive change to add the feature that I actually need, so if this project is maintenance-only then I'll implement what I need as a standalone codebase rather than as part of `x11vnc`.

Feel free to merge or discard this PR as you see fit.
